kernelci/api/models.py

```diff
@@ -117,7 +120,8 @@ class Revision(BaseModel):
     tree: str = Field(
         description="git tree of the revision"
     )
-    url: Annotated[AnyUrl | FileUrl, AfterValidator(str)] = Field(
+    url: Annotated[str, BeforeValidator(
+        lambda value: str(any_url_adapter.validate_python(value)))] = Field(
```

Comment:
Support for [FileURLs without host set](https://docs.pydantic.dev/latest/api/networks/#pydantic.networks.FileUrl) is dropped here but I don't think any kind of currently supported Storage types allowed them anyway

Reply:
`AnyUrl` allows `FileUrl` as well. That's why dropped it from here.